dd-sdk-android-gradle-plugin/src/main/kotlin/com/datadog/gradle/plugin/DdSymbolFileUploadTask.kt

```diff
@@ -0,0 +1,159 @@
+package com.datadog.gradle.plugin
+
+import com.android.build.gradle.api.ApplicationVariant
+import com.android.build.gradle.tasks.ExternalNativeBuildTask
+import com.datadog.gradle.plugin.internal.ApiKey
+import com.datadog.gradle.plugin.internal.Uploader
+import org.gradle.api.Project
+import org.gradle.api.file.ConfigurableFileCollection
+import org.gradle.api.file.DirectoryProperty
+import org.gradle.api.logging.Logging
+import org.gradle.api.model.ObjectFactory
+import org.gradle.api.provider.ProviderFactory
+import org.gradle.api.tasks.InputFiles
+import org.gradle.api.tasks.TaskProvider
+import java.io.File
+import javax.inject.Inject
+import kotlin.reflect.full.memberProperties
+
+/**
+ * A Gradle task to upload NDK symbol files to Datadog servers.
+ */
+internal abstract class DdSymbolFileUploadTask @Inject constructor(
+    objects: ObjectFactory,
+    providerFactory: ProviderFactory
+) : DdFileUploadTask(providerFactory) {
+
+    @get:InputFiles
+    val searchDirectories: ConfigurableFileCollection = objects.fileCollection()
+
+    init {
+        description =
+            "Uploads NDK symbol files to Datadog servers to perform native crash symbolication."
+    }
+
+    override fun getFilesList(): List<Uploader.UploadFileInfo> {
+        val files = mutableListOf<Uploader.UploadFileInfo>()
+
+        searchDirectories
+            .flatMap(this::findSoFiles)
+            .toSortedSet(compareBy { it.absolutePath })
+            .toSet()
+            .forEach { file ->
+                val arch = file.parentFile.name
+                require(SUPPORTED_ARCHS.contains(arch))
```

Comment:
It was, but now I'm reconsidering.

Reply:
Yeah, I think we should leave this in. If we're getting a weird ARCH we probably want to fast fail so we get a bug report.

Reply:
As long as the exception is thrown during task execution phase (and it is), and not task configuration phase, I'm fine.

The difference is that task configuration phase is done with other tasks configuration. So if there is an error during the configuration phase, other plugins/tasks won't be configured (if I'm not wrong) as well, this will have big impact.

Task execution phase on another hand is independent (well, if it has no other tasks dependent), so error during the execution has much smaller impact.